jhdf/src/main/java/io/jhdf/WritableGroupImpl.java

```diff
@@ -41,151 +41,206 @@
-		}
-		WritableGroupImpl newGroup = new WritableGroupImpl(this, name);
-		children.put(name, newGroup);
-		logger.info("Added group [{}] to group [{}]", name, getPath());
-		return newGroup;
-	}
-
-	@Override
-	public Iterator<Node> iterator() {
-		Collection<Node> values = Collections.unmodifiableCollection(children.values());
-		return values.iterator();
-	}
-
-	public long write(HdfFileChannel hdfFileChannel, long position) {
-		logger.info("Writing group [{}] at position [{}]", getPath(), position);
-
-		List<Message> messages = new ArrayList<>();
-		GroupInfoMessage groupInfoMessage = GroupInfoMessage.createBasic();
-		messages.add(groupInfoMessage);
-		LinkInfoMessage linkInfoMessage = LinkInfoMessage.createBasic();
-		messages.add(linkInfoMessage);
-
-		for (Map.Entry<String, WritableNode> child : children.entrySet()) {
-			LinkMessage linkMessage = LinkMessage.create(child.getKey(), 0L);
-			messages.add(linkMessage);
-		}
-
-		if(!getAttributes().isEmpty()) {
-			// Need an attribute info message to allow HDFView to see the attributes
-			AttributeInfoMessage attributeInfoMessage = AttributeInfoMessage.create();
-			messages.add(attributeInfoMessage);
-			for (Map.Entry<String, Attribute> attribute : getAttributes().entrySet()) {
-				logger.info("Writing attribute [{}] in group [{}]", attribute.getKey(), getName());
-				AttributeMessage attributeMessage = AttributeMessage.create(attribute.getKey(), attribute.getValue());
-				messages.add(attributeMessage);
-			}
-		}
-
-		ObjectHeader.ObjectHeaderV2 objectHeader = new ObjectHeader.ObjectHeaderV2(position, messages);
-
-		ByteBuffer tempBuffer = objectHeader.toBuffer();
-		int objectHeaderSize = tempBuffer.limit();
-		// Upto here just finding out the size of the OH can be much improved
-
-		// Start building another OH
-		messages = new ArrayList<>();
-		messages.add(groupInfoMessage);
-		messages.add(linkInfoMessage);
-
-		if(!getAttributes().isEmpty()) {
-			AttributeInfoMessage attributeInfoMessage = AttributeInfoMessage.create();
-			messages.add(attributeInfoMessage);
-			for (Map.Entry<String, Attribute> attribute : getAttributes().entrySet()) {
-				logger.info("Writing attribute [{}]", attribute.getKey());
-				AttributeMessage attributeMessage = AttributeMessage.create(attribute.getKey(), attribute.getValue());
-				messages.add(attributeMessage);
-			}
-		}
-
-		long nextChildAddress = position + objectHeaderSize;
-
-		for (Map.Entry<String, WritableNode> child : children.entrySet()) {
-			LinkMessage linkMessage = LinkMessage.create(child.getKey(), nextChildAddress);
-			messages.add(linkMessage);
-			long endPosition = child.getValue().write(hdfFileChannel, nextChildAddress);
-			nextChildAddress = endPosition;
-		}
-
-		objectHeader = new ObjectHeader.ObjectHeaderV2(position, messages);
-		hdfFileChannel.write(objectHeader.toBuffer(), position);
-
-		logger.info("Finished writing group [{}]", getPath());
-		return nextChildAddress;
-	}
```

Comment:
Think this is some formatting issue?

Reply:
yes, I'll revert the format change. (IDE auto-format that I missed)

Reply:
done